x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistry.java

```diff
@@ -405,6 +405,17 @@ public EsqlFunctionRegistry snapshotRegistry() {
         return snapshotRegistry;
     }
 
+    public boolean isSnapshotOnly(String functionName) {
```

Comment:
I think this can be `static` and it'd make things a little less confusing if it were.